src/tuf.rs

```diff
@@ -190,14 +189,12 @@ impl<D: DataInterchange> Tuf<D> {
             //     discard it, abort the update cycle, and report the rollback attack. On the next
             //     update cycle, begin at step 0 and version N of the root metadata file.
 
-            // Next, make sure the new root has a higher version than the old root.
-            if new_root.version() == trusted_root.version() {
-                info!(
-                    "Attempted to update root to new metadata with the same version. \
-                     Refusing to update."
-                );
-                return Ok(false);
-            } else if new_root.version() < trusted_root.version() {
+            let next_root_version = trusted_root
+                .version()
+                .checked_add(1)
+                .expect("root version should be less than max u32");
```

Comment:
should we error instead of panic here?

semi-related, 2*32-1 doesn't seem very high for a process that could be automated

Reply:
Sure, I'll switch. We'd never trip over this in practice though because the new root would fail to deserialize a 2^32 value into a u32.

We could bump the version numbers up to a u64 are worthwhile, but consider the case where it only took 1 millisecond to create or fetch a root metadata. Trying to generate or fetch `(2^32-1) - N` would still take approximately 50 days. Repositories and clients should instead have separate monitoring in place to catch runaway processes like that.